### DISCUSSION TITLE
Audio Player: Programmatically change source

### DISCUSSION THREAD

**Discussion by fakerybakery** [vidstack/player#1380] [Q&A]
Hi,
How can I programmatically change the source of a Vidstack audio player? I'm using Vanilla JS/CDN.
Thanks!

**Comment by mihar-22** [comment]
Simply set the `player.src` to the new source. Most of the information you need should be found in the docs.

**Reply by fakerybakery** [reply]
Thanks - I'm trying to change it to an uploaded audio file in an `input type=file` element. I'm using this code:
```javascript
const player = document.querySelector('media-player');
player.src = URL.createObjectURL(document.getElementById('file').files[0]);
```
But the player doesn't change - instead, it disappears from the page. It works on the HTML audio element though

**Reply by fakerybakery** [reply]
Actually, this seems to only be an issue when getting the DOM element, not when I create the player programatically. Is there any way to change it for an existing player?

**Reply by mihar-22** [reply]
You need to provide the source type so the player knows which provider to load. In this case you can do:

```ts
player.src = {
  src: URL.createObjectURL(...),
  type: 'audio/mp3',
}
```
